crates/refunder/src/refund_service.rs

```diff
@@ -170,43 +157,65 @@ impl RefundService {
         Ok(order_to_ethflow_data(order, ethflow_order))
     }
 
-    async fn send_out_refunding_tx(&mut self, uids: Vec<OrderUid>) -> Result<()> {
+    async fn send_out_refunding_tx(
+        &mut self,
+        uids: Vec<(OrderUid, Vec<CoWSwapEthFlowAddress>)>,
+    ) -> Result<()> {
         if uids.is_empty() {
             return Ok(());
         }
-        // only try to refund MAX_NUMBER_OF_UIDS_PER_REFUND_TX uids, in order to fit
-        // into gas limit
-        let uids: Vec<OrderUid> = uids
-            .into_iter()
-            .take(MAX_NUMBER_OF_UIDS_PER_REFUND_TX)
-            .collect();
-
-        tracing::debug!("Trying to refund the following uids: {:?}", uids);
-
-        let futures = uids.iter().map(|uid| {
-            let (uid, self_) = (*uid, &self);
-            async move {
-                self_
-                    .get_ethflow_data_from_db(&uid)
-                    .await
-                    .context(format!("uid {uid:?}"))
+
+        // Go over all order uids and group them by ethflow contract address
+        let mut uids_by_contract: HashMap<CoWSwapEthFlowAddress, Vec<OrderUid>> = HashMap::new();
```

Comment:
This could probably be replace with [Itertools::group_by](https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.group_by)